yarn-project/ethereum/src/deploy_l1_contracts.ts

```diff
@@ -635,7 +635,8 @@ export const deployRollup = async (
       });
       logger.verbose(`Adding rollup ${rollupContract.address} to GSE ${addresses.gseAddress} in tx ${addRollupTxHash}`);
 
-      txHashes.push(addRollupTxHash);
+      // wait for this tx to land in case we have to register initialValidators
+      await extendedClient.waitForTransactionReceipt({ hash: addRollupTxHash });
```

Comment:
This is the actual fix. Five lines below we execute

```
if (args.initialValidators && (await gseContract.read.isRollupRegistered([rollupContract.address]))) {
  await addMultipleValidators(
```

Which I don't think will work if the this tx isn't mined